### PR TITLE
storage: Checkboxes for NFS mount options

### PR DESCRIPTION
**Pull request by mvollmer**
- [x] Tests
- [x] #7750 
- [x] #7580

### CODE REVIEW

**Comment by andreasn** [comment]
looks good to me!

**Comment by martinpitt** [comment]
Semaphore build fails:

    /tmp/source/pkg/storaged/overview.js:545:58: Use '===' to compare with ''.

File: test/verify/check-storage-nfs [hunk 1]
```diff
@@ -80,6 +80,17 @@ class TestStorage(StorageCase):
...
+        # Make /home/bar read-only
+        b.click("#nfs-mounts .nfs-arm-button")
+        b.click("#nfs-mounts button[data-action=mount-edit][data-arg=1]")
+
+        self.dialog_wait_open()
+        self.dialog_set_val("mount_ro", True)
+        self.dialog_apply()
+        self.dialog_wait_close()
+
+        self.assertEqual (m.execute("findmnt -s -n -o OPTIONS /mounts/barbar").strip(), "ro")
```
Comment (martinpitt) [inline test/verify/check-storage-nfs]: Would be nice to also check the `defaults`, `noauto`, and custom mount options here, to cover all code paths.

File: pkg/storaged/overview.js [hunk 2]
```diff
@@ -476,6 +478,23 @@
                 return this_deferred.promise();
             }
 
+            var mount_options = entry? entry.fields[3] : "defaults";
+            var split_options = format.parse_options(mount_options == "defaults" ? "" : mount_options);
+            var opt_auto = !format.extract_option(split_options, "noauto");
+            var opt_ro = format.extract_option(split_options, "ro");
```
Comment (martinpitt) [inline pkg/storaged/overview.js]: To  be sure: `defaults` is just a "visible" name for an empty set of mount options -- i. e. if the mount options are `ro`, then the implied default is still `auto`, right?

Comment (mvollmer) [reply pkg/storaged/overview.js]: Yes.

File: test/verify/check-storage-nfs (hunk 1 above)
Comment (mvollmer) [reply test/verify/check-storage-nfs]: Done.

**Comment by mvollmer** [comment]
@martinpitt, ptalthx.

**Comment by martinpitt** [comment]
Screenshot for release notes:

storage-nfs-option-checkboxes

**Pull request merged**